src/behavior-considered-undefined.md

```diff
@@ -151,10 +155,30 @@ entire range, so it is important that the length metadata is never too large. In
 particular, the dynamic size of a Rust value (as determined by `size_of_val`)
 must never exceed `isize::MAX`.
 
+### Unwinding inappropriately
+[unwinding]: #unwinding-inappropriately
+
+Unwinding with the wrong [ABI][abi] is undefined behavior:
+
+* Causing an unwind into Rust code from a foreign function that was called via a
+  function declaration or pointer declared with a non-unwinding ABI, such as `"C"`,
+  `"system"`, etc. (For example, this case occurs when such a function written in
+  C++ throws an exception that is uncaught and propagates to Rust.)
+* Calling a Rust `extern` function that unwinds (with `extern "C-unwind"` or
+  another ABI that permits unwinding) from a runtime that does not support.
+  unwinding, such as code compiled with GCC or Clang using `-fno-exceptions`
+
+Catching an unwinding operation in a different runtime than the one in which it
+initiated is also undefined behavior:
+
+* Catching a Rust `panic` in non Rust code (for instance `catch (...)` in C++).
+* Catching a non-Rust unwind (such as a C++ exception) in Rust (that is, with `catch_unwind`).
```

Comment:
This should also mention that catching a Rust `panic` with a `catch_unwind` in a different Rust runtime (including different staticlib/cdylibs) is undefined.

Reply:
@chorman0773 How about:

> Catching a Rust `panic` in non Rust code (for instance `catch (...)` in C++), or in Rust code compiled or linked with a different runtime.

I don't think it would be helpful here to link to `runtime.md`, since that doesn't really define what constitutes the runtime implementation, and the first section is about the panic handler, which may give the incorrect impression that differences in the panic handler are the root cause of incompatibility between runtimes.